### ISSUE TITLE
Load the LSP from workspace in language extension

### ISSUE THREAD

**Issue by chungweileong94** [zed-industries/zed#10760]
### Check for existing issues

- [X] Completed

### Describe the feature

Some extension needs to load the LSP installed in the workspace, if not fallback to the global installed LSP.

**For example:**
The [biome extension](https://github.com/biomejs/biome-zed) needs to be able to load the LSP binary installed in the workspace, if not install and fallback to a global LSP binary.

However, it doesn't seem like we are able to check if an LSP binary exists in the workspace, as things like `Path::exists` don't really work (probably because of wasm?). 

I also tried to use `worktree.read_text_file()` to see if the `node_modules/@biomejs/biome/package.json` exists in the workspace, it works sometimes but mostly doesn't work. https://github.com/biomejs/biome-zed/pull/9

**Comment by jansol** [comment]
AIUI this is also necessary for Android development where you need a project-specific version of gradle.

**Comment by xcrap** [comment]
Language server biome:

failed to spawn command. path: "/Users/xcrap/Library/Application Support/Zed/extensions/work/biome/node_modules/@biomejs/cli-darwin-arm64/biome", working directory: "/Users/xcrap/www/goshapenutrition", args: ["lsp-proxy"]
-- stderr--

My biome stopped working at all in both Zed and Zed Preview, it's the same issue?

**Comment by chungweileong94** [comment]
> Language server biome:
> 
> failed to spawn command. path: "/Users/xcrap/Library/Application Support/Zed/extensions/work/biome/node_modules/@biomejs/cli-darwin-arm64/biome", working directory: "/Users/xcrap/www/goshapenutrition", args: ["lsp-proxy"] -- stderr--
> 
> My biome stopped working at all in both Zed and Zed Preview, it's the same issue?

Not really, but I would suggest you to file an issue on biome-zed repo https://github.com/biomejs/biome-zed/issues. 

This issue is more like an extension API feature request. But due to the lack of APIs, the workaround that the biome extension used might cause the issue that you are facing.
